Week3/homework/part1/index.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+{
+  // root div declaration
+  const root = document.getElementById('root');
+
+  // this function will fetch JSON data using async/await and axios
+  async function fetchJSON(url) {
+
+    try {
+
+      const response = await axios.get(url);
+      // throw the error if exists
+      if (!response.statusText) {
+        throw new Error(`Network error: ${response.status},${response.statusText}`);
+      }
+      const resData = await response.data;
+      // return the fetched data
+      return resData;
+
+    } 
+    catch (error) {
+      // catch the error if exists
+      printError(error);
+
+    }
+  }
+
+  // this function create and append elements
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.entries(options).forEach(([key, value]) => {
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  // this function render the selected Repository details
+  function renderRepoDetails(repo, repoContainer) {
+    // create Repository table
+    const table = createAndAppend('table', repoContainer);
+
+    // create Repository title row
+    const titleTr = createAndAppend('tr', table);
+    createAndAppend('td', titleTr, { text: 'Repository:', class: 'title' });
+    const td = createAndAppend('td', titleTr);
+    createAndAppend('a', td, {
+			href: repo.html_url,
+			text: repo.name,
+			target: '_blank'
+		});
+
+    // create Description row
+    const descriptionTr = createAndAppend('tr', table);
+    createAndAppend('td', descriptionTr, { text: 'Description:', class: 'title' });
+    createAndAppend('td', descriptionTr, { text: repo.description });
+
+    // create Fork row
+    const forksTr = createAndAppend('tr', table);
+    createAndAppend('td', forksTr, { text: 'Forks:', class: 'title' });
+    createAndAppend('td', forksTr, { text: repo.forks });
+
+    // create Updated row
+    const updatedTr = createAndAppend('tr', table);
+    createAndAppend('td', updatedTr, { text: 'Updated:', class: 'title' });
+    createAndAppend('td', updatedTr, { text: repo.updated_at });
+  }
+
+  // this function render the Contributions details
+  async function renderContributorsDetails(repo, contributorsContainer) {
+    // create section title
+    const ul = createAndAppend('ul', contributorsContainer, {
+      text: 'Contributions',
+      class: 'contributors-ul' 
+    });
+
+    // fetch contributors data 
+    try {
+      const contributors = await fetchJSON(repo.contributors_url);
+      // create contributors list
+      contributors.forEach( Contributor => {
```

Comment:
keep your `C` low, It's a common thing we have to keep it camelCase for normal difinition and PascalCase for constructors to differentiate.
```suggestion
      contributors.forEach( contributor => {
```
update the rest :)